reference/docs-conceptual/samples/Creating-Get-WinEvent-queries-with-FilterHashtable.md

```diff
@@ -0,0 +1,250 @@
+
+This article presents information about how to use enumerated values in a hash table. For more
+information about enumeration, read these **Scripting Guy** blog posts. To create a function that
+returns the enumerated values, see
+[Enumerations and Values](https://devblogs.microsoft.com/scripting/hey-scripting-guy-weekend-scripter-enumerations-and-values).
+For more information, see the
+[Scripting Guy series of blog posts about enumeration](https://devblogs.microsoft.com/scripting/?s=about+enumeration).
+
+### Hash table key/value pairs
+
+To build efficient queries, use the `Get-WinEvent` cmdlet with the **FilterHashtable** parameter.
+**FilterHashtable** accepts a hash table as a filter to get specific information from Windows event
+logs. A hash table uses **key/value** pairs. For more information about hash tables, see [about_Hash_Tables](/powershell/module/microsoft.powershell.core/about/about_hash_tables).
+
+This sample uses several of the **FilterHashtable** parameter's **key/value** pairs. The completed
+query includes **LogName**, **ProviderName**, **Keywords**, **ID**, and **Level**.
+
+The accepted **key/value** pairs are shown in the following table and are included in the
+documentation for the [Get-WinEvent](/powershell/module/microsoft.powershell.diagnostics/Get-WinEvent)
+**FilterHashtable** parameter.
+
+The following table displays the key names, data types, and whether wildcard characters are accepted
+for a data value.
+
+| Key name     | Value data type    | Accepts wildcard characters? |
+|------------- | ------------------ | ---------------------------- |
+| LogName      | `<String[]>`       | Yes |
+| ProviderName | `<String[]>`       | Yes |
+| Path         | `<String[]>`       | No  |
+| Keywords     | `<Long[]>`         | No  |
+| ID           | `<Int32[]>`        | No  |
+| Level        | `<Int32[]>`        | No  |
+| StartTime    | `<DateTime>`       | No  |
+| EndTime      | `<DateTime>`       | No  |
+| UserID       | `<SID>`            | No  |
+| Data         | `<String[]>`       | No  |
+| *            | `<String[]>`       | No  |
+
+### Building a query with a hash table
+
+To verify results and troubleshoot problems, it helps to build the hash table one **key/value** pair
+at a time. The query gets data from the **Application** log. The hash table is equivalent to
+`Get-WinEvent –LogName Application`.
+
+To begin, create the `Get-WinEvent` query. Use the **FilterHashtable** parameter's **key/value**
+pair with the key, **LogName**, and the value, **Application**.
+
+```PowerShell
+Get-WinEvent -FilterHashtable @{ LogName='Application' }
+```
+
+Continue to build the hash table with the **ProviderName** key. The **ProviderName** is the name
+that appears in the **Source** field in the **Windows Event Viewer**. For example, **.NET Runtime**
+in the following screenshot:
+
+![Image of Windows Event Viewer sources.](/media/creating-get-winEvent-queries-with-filterhashtable/providername.png)
```

Comment:
broken link - the correct link should be:

`./media/creating-get-winEvent-queries-with-filterhashtable/providername.png`

Reply:
Thanks, resolved.